### ISSUE TITLE
`save("sound.wav", buf)` seems to write with the wrong samplerate

### ISSUE THREAD

**Issue by anandijain** [JuliaAudio/PortAudio.jl#88]
```julia 
using PortAudio, SampledSignals, LibSndFile, FileIO
devs = PortAudio.devices()
mic = PortAudioStream("MacBook Pro Microphone", 1, 0)
buf = read(mic, 2s)
close(mic)
save("data/grooovin2.wav", buf)
```
This writes the wav file at 8khz for me. I'm not sure why, considering `buf.samplerate == 48000`

**Comment by anandijain** [comment]
Is there an option to specify samplerate that I'm missing?

**Comment by bramtayl** [comment]
Huh that's weird. Yes, there is an optional `samplerate` keyword argument that you can pass to `PortAudioStream`. But it should default to using your device defaults. If `buf` is correct, then I think the issue is probably with `SampledSignals`, unless I'm missing something.

**Comment by rob-luke** [comment]
I ran this code and was unable to reproduce your issue. Are you using some validated tool to check the wav file like audition or QuickTime etc? Either way, I don't think this library supports `save`, isn't that LibSndFile?

**Comment by anandijain** [comment]
hmm. see #87 for the `save` issues i hit. 

yes both Ableton and WAV.jl are giving me 8kHz. (in that issue I neglected to put in the `using WAV`, which is where `wavread` comes from).

it could be possible its a strange M1 issue. 

as you can see in that issue i opened in sampledsignals, that I have to use my own jll's while i wait on my yggdrasil PRs to get merged

**Comment by rob-luke** [comment]
> it could be possible its a strange M1 issue.

I hope not, that sounds hard to debug.

Yeah, thanks for opening these issues. Please keep us posted as you dig deeper in to the issue. I can also run snippets you provide on my intel Mac if that helps.

**Comment by bramtayl** [comment]
Oops sorry my bad, save is probably an issue with LibSndFile, not SampledSignals

**Comment by sprague252** [comment]
> it could be possible its a strange M1 issue.

This is not an M1 issue. I have the same problem on my Intel MacBook Pro [2.3 GHz Quad-Core Intel Core i7, Mac OS 12.3.1 (21E258), Julia 1.7.2 (2022-02-06), PortAudio v1.2.0, LibSndFile v2.4.0, SampledSignals v2.1.3]. When I write a buffer recorded from my internal microphone at 48000.0 Hz to a WAV file, the WAV file header says the sample rate is 8000 Hz. If I change the buffer `samplerate`, it still writes a file with an 8000 Hz sample rate. If I write an OGG file, there is no problem.  I suspect that somewhere the buffer `samplerate` field is not getting written to the WAV file sample rate. Here is some code that produces the issue. The parameter `buf` is a 10 second buffer recorded from my internal microphone at 48000.0 Hz.

```
julia> buf
480000-frame, 1-channel SampleBuf{Float32, 2}
10.0s sampled at 48000.0Hz
▆▆▆▆▆▆▅▄▄▅▆▆▆▅▅▄▄▄▅▆▆▅▆▆▄▆▆▅▄▅▄▄▄▅▆▆▆▅▅▄▄▄▆▆▆▆▅▅▆▆▅▄▄▄▄▄▆▆▆▅▄▄▄▅▆▅▅▆▅▄▅▆▅▄▄▄▄▄▄▄

julia> buf.samplerate
48000.0

julia> save(joinpath(homedir(), "Desktop", "myvoice.wav"), buf)

julia> save(joinpath(homedir(), "Desktop", "myvoice.ogg"), buf)
```
The resulting WAV file has a sample rate of 8000 Hz, and the OGG file has no issues.

**Comment by ssfrr** [comment]
Do you have WAV.jl installed? It could be that FileIO is dispatching to that, which doesn’t know how to interpret SampleBuf sample rates. Try in a clean environment.

**Comment by sprague252** [comment]
Yes, I have WAV.jl installed. In fact, I can save WAV files with WAV.jl with no issues. I will try a clean environment and report back.
